### DISCUSSION TITLE
Jupyter extension starts multiple idle R kernels, clogging up memory

### DISCUSSION THREAD

**Discussion by pehkawn** [microsoft/vscode-jupyter#9409] [Questions and Answers]
I've been running R using the Jupyter notebook extension for some time, and for the most part it's working great. I have an issue with how it chews up my memory however. In my system monitor, I can see multiple idle R kernels loaded in memory. They use 0% of CPU, indicating they're just just there without doing anything. During extensive use, the amount of R kernels will keep growing until it takes up all my available memory. Running garbage collect (`gc()`) or clearing up variables `rm(list = ls() )` will do nothing to free up memory, and the only way to clear up my memory is to terminate the process. This  I can not think this is working after intention, but I am uncertain whether this is a bug, or the matter of changing a setting. Any input on this would be appreciated.

#### Version info:
**VS Code:** 1.65.2
**Jupyter extension:** v2022.2.1030672458
**R:** 4.1.3
**Anaconda:** 4.11.0
**OS:** Ubuntu 20.04.4 LTS

![Skjermbilde fra 2022-03-17 15-19-12](https://user-images.githubusercontent.com/44672121/158901540-277b2964-14a5-4ce9-8188-7a26a9457150.png)

**Comment by IanMatthewHuff** [comment]
@pehkawn Could you pull the contents of the Jupyter tab of the output window maybe after running a few R notebook files and then closing them (the output can be pretty big, so saving in a file and attaching that might help). In general kernels should be shutting down in VS Code the same way that they would with Jupyter notebook. Output logging might show if something is failing during this shutdown process.

Also I wasn't quite sure, are these kernels being spawned just via working with one file? Or are you opening and closing multiple R notebook files in your workflow?

**Reply by pehkawn** [reply]
Ok, I've attached the [output from my most recent session](https://github.com/microsoft/vscode-jupyter/files/8295803/console_output.txt), which has been fairly short (~20min). Already there are 23 separate kernels in memory. 
To answer you questions: Yes, these kernels are spawned from working with one notebook file, and not from opening and closing multiple notebook files.

**Reply by IanMatthewHuff** [reply]
Hmmm, there is some output coming in from stderr on the R kernel. But looking at it, it looks like just some warnings and info messages from the R kernel starting, nothing that should be creating or killing an existing kernel. Question can you also run this kernel from the `jupyter notebook` web interface? From vscode's perspective it looks like we are just starting one instance of the R kernel, it might be that kernel is starting up more instances under the covers somehow. If the standard jupyter notebook book is experiencing the same issue that would narrow it down some.

**Comment by greazer** [comment]
Closing in favor of tracking as a bug: https://github.com/microsoft/vscode-jupyter/issues/9700
